### PR TITLE
Added support for clustered box plots

### PR DESCRIPTION
**Pull request by jacoblurye**
Group box plots above shared x axis labels.

screen shot 2018-08-03 at 2 36 21 pm

### CODE REVIEW

File: shared/src/main/scala/com/cibo/evilplot/numeric/BoxPlotSummaryStatistics.scala [hunk 1]
```diff
@@ -42,7 +42,8 @@ case class BoxPlotSummaryStatistics(
   middleQuantile: Double,
   upperQuantile: Double,
   outliers: Seq[Double],
-  allPoints: Seq[Double]) {
+  allPoints: Seq[Double],
+  cluster: Int) {
```
Comment (zacharygraziano) [inline shared/src/main/scala/com/cibo/evilplot/numeric/BoxPlotSummaryStatistics.scala]: I feel as if this is just a container for the results of the box plot summary stuff, and it probably shouldn't contain the cluster number (I think there is a `BoxRendererContext` or similar that I'd be more comfortable using for stuff like this)

File: shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala [hunk 2]
```diff
@@ -98,6 +125,67 @@ object BoxPlot {
...
+  def clustered(
+    data: Seq[Seq[Seq[Double]]],
+    quantiles: (Double, Double, Double) = (0.25, 0.50, 0.75),
+    spacing: Option[Double] = None,
+    clusterSpacing: Option[Double] = None,
+    boundBuffer: Option[Double] = None,
+    boxRenderer: Option[BoxRenderer] = None,
+    pointRenderer: Option[PointRenderer] = None
+  )(implicit theme: Theme): Plot = {
+    println(data.length)
```
Comment (zacharygraziano) [inline shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala]: println 🚶

File: shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala [hunk 3]
```diff
@@ -98,6 +125,67 @@ object BoxPlot {
...
+    * @param data the distributions to plot
+    * @param boxRenderer the `BoxRenderer` to use to display each distribution
+    * @param pointRenderer the `PointRenderer` used to display outliers
+    * @param quantiles quantiles to use for summary statistics.
+    *                  defaults to 1st, 2nd, 3rd quartiles.
+    * @param spacing spacing how much spacing to put between boxes
+    * @param boundBuffer expand bounds by this factor
+    */
+  def custom(
+    summaries: Seq[Option[BoxPlotSummaryStatistics]],
```
Comment (zacharygraziano) [inline shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala]: why are we using `Seq[Option[BoxPlotSummaryStatistics]]` here? what happens with `None`s?

File: shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala [hunk 4]
```diff
@@ -98,6 +125,67 @@ object BoxPlot {
...
+    boxRenderer: Option[BoxRenderer],
```
Comment (zacharygraziano) [inline shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala]: did you mean to the default these to `None`?

File: shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala [hunk 5]
```diff
@@ -113,28 +201,9 @@ object BoxPlot {
...
-    * @param boxRenderer the `BoxRenderer` to use to display each distribution
-    * @param pointRenderer the `PointRenderer` used to display outliers
-    * @param quantiles quantiles to use for summary statistics.
-    *                  defaults to 1st, 2nd, 3rd quartiles.
-    * @param spacing spacing how much spacing to put between boxes
-    * @param boundBuffer expand bounds by this factor
-    */
-  @deprecated("use apply", "2018-03-28")
-  def custom(
-    data: Seq[Seq[Double]],
```
Comment (zacharygraziano) [inline shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala]: depending on whether we release this as part of 0.5.0 or 0.4.2 we may want/need to keep this for now.

File: shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala [hunk 6]
```diff
@@ -98,6 +125,67 @@ object BoxPlot {
...
+    *
+    * @param data the distributions to plot
+    * @param boxRenderer the `BoxRenderer` to use to display each distribution
+    * @param pointRenderer the `PointRenderer` used to display outliers
+    * @param quantiles quantiles to use for summary statistics.
+    *                  defaults to 1st, 2nd, 3rd quartiles.
+    * @param spacing spacing how much spacing to put between boxes
+    * @param boundBuffer expand bounds by this factor
+    */
+  def custom(
```
Comment (triton11) [inline shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala]: I don't think un-deprecating it breaks anything, but I am curious if there is a protocol for that kinda change. Do we need a `custom` and an `apply`? Or could we merge the two and default the extra custom parameters to `None`?

**Pull request merged**